posts/2021-08-13-dev-mode-multi-module.adoc

```diff
@@ -0,0 +1,130 @@
+
+== Running Liberty on a multi-module project
+
+Liberty can be run on a multi-module project with a single command by using dev mode, dev mode with container support, or by starting the server in the foreground.  To do any of these, the Liberty Maven plug-in must first be defined.
+
+=== Defining the Liberty Maven plug-in
+
+To use the Liberty Maven plug-in in your multi-module project, you can define the plug-in in either the parent `pom.xml` file of every module or in the `pom.xml` of every module.
+
+For example, if your modules have a `<parent>` element that points to a parent project, add the Liberty Maven plug-in to the `<pluginManagement>` section of the parent project's `pom.xml` as follows:
+[source,xml]
+----
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.openliberty.tools</groupId>
+                    <artifactId>liberty-maven-plugin</artifactId>
+                    <version>3.4</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+----
+
+=== Starting dev mode
+
+To start dev mode, from the directory containing the top-level multi-module `pom.xml`, run:
+
+----
+mvn liberty:dev
+----
+
+Once dev mode starts successfully, changes in all modules are detected and hot deployed. By default, any dependent modules will also be recompiled and deployed. For example, if a change is made in the `jar` module of the sample application described above, the `war` module will also be recompiled since it depends on the `jar` module. 
+
+**gif of change made to jar module, show compilation of jar and war module**
+
+=== Starting dev mode with container support
+
+There is container support for multiple module projects and dev mode. Use the `devc` goal to start dev mode with container support:
+
+----
+mvn liberty:devc  
+----
+
+If your Dockerfile is not in the same directory as where you started the dev mode command from, you can point to it with the `-Ddockerfile=<path-to-dockerfile>` paremeter. For more information on Dockerfiles for your Open Liberty application, see the link:https://github.com/OpenLiberty/ci.docker#building-an-application-image[Open Liberty Docker image documentation]. 
+
+=== Running tests
+You can run tests on all modules by pressing Enter. 
+
+**gif of enter to run tests**
+
+You can also enable hot tests so that tests will run on every code change. To enable hot tests, add the `-DhotTests` parameter: `mvn liberty:dev -DhotTests`. With hot tests enabled, tests will run on the module that was changed and any dependent modules.
+
+=== Stopping dev mode
+
+You can exit dev mode by pressing Control-C, or type `q` and press Enter in the terminal.
```

Comment:
To exit dev mode, by press Control+C or by type `q` in the terminal and press Enter.